### PR TITLE
feat(home-manager): init fzf module

### PR DESCRIPTION
**Pull request by Anomalocaridid**
Adds support for fzf.

I took the same approach that the console module uses and used the palette to manually add the colors from [catppuccin/fzf](https://github.com/catppuccin/fzf).

### CODE REVIEW

File: modules/home-manager/fzf.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    lib.ctp.mkCatppuccinOpt "fzf";
+
+  config.programs.fzf.colors = lib.mkIf enable
+    # Manually populate with colors from catppuccin/fzf
+    (lib.attrsets.mapAttrs (_: color: palette.${color}.hex)
+      {
+        "bg+" = "surface0";
+        bg = "base";
+        spinner = "rosewater";
+        hl = "red";
```
Comment (isabelroses) [inline modules/home-manager/fzf.nix:22]: nit: group hl with hl+ and the same for fg and fg+ like you did for bg and bg+

Comment (Anomalocaridid) [reply modules/home-manager/fzf.nix:22]: I can do that, but first I should mention I put the options in that specific order to match the (seemingly arbitrary) order of the options in catppuccin/fzf to make it easier to compare.

Comment (isabelroses) [reply modules/home-manager/fzf.nix:22]: Ahhh, that makes total sense.

**Comment by getchoo** [comment]
this needs a rebase

**Comment by Anomalocaridid** [comment]
Rebased

**Review by getchoo — APPROVED** [review]
thanks!

**Pull request merged**